### PR TITLE
fix "Use this address..." string?

### LINKED ISSUE #20054: Some enhancements to a string?
**Describe the solution you'd like**
enhance string

**Describe alternatives you've considered**
nothing currently

**Additional context**
The string `'Use this address to <a href="%s" target="_blank" rel="noreferrer noopener">access your Files via WebDAV</a>' in `https://github.com/nextcloud/server/blob/2efa00070571073c6b4118afedfd632321c7fb39/apps/files/templates/appnavigation.php#L46
should better be called `'<a href="%s" target="_blank" rel="noreferrer noopener">Use this address to access your Files via WebDAV</a>'`, shouldn't it?

### PR DESCRIPTION
**Pull request by szaimen**
fix https://github.com/nextcloud/server/issues/20054?

I just wanted to bring this change in to start the discussion on this topic.
Any thoughts on this?

### CODE REVIEW

File: apps/files/templates/appnavigation.php [hunk 1]
```diff
@@ -43,7 +43,7 @@
 			<label for="webdavurl"><?php p($l->t('WebDAV')); ?></label>
 			<input id="webdavurl" type="text" readonly="readonly"
 				   value="<?php p($_['webdav_url']); ?>"/>
-			<em><?php print_unescaped($l->t('Use this address to <a href="%s" target="_blank" rel="noreferrer noopener">access your Files via WebDAV</a>', array(link_to_docs('user-webdav')))); ?></em>
+			<em><?php print_unescaped($l->t('<a href="%s" target="_blank" rel="noreferrer noopener">Use this address to access your Files via WebDAV</a>', array(link_to_docs('user-webdav')))); ?></em>
```
Comment (rullzer) [inline apps/files/templates/appnavigation.php]: As long as we are fixing things. Could you move out the `a` parts of the translation?

Comment (szaimen) [reply apps/files/templates/appnavigation.php]: @rullzer Do you mean something like this?
```suggestion
			<em><?php print_unescaped(<a href="%s" target="_blank" rel="noreferrer noopener">, $l->t('Use this address to access your Files via WebDAV'), </a>, array(link_to_docs('user-webdav'))); ?></em>
```

Comment (rullzer) [reply apps/files/templates/appnavigation.php]: Yes. No need to have our translaters translate the link stuff :)

Comment (szaimen) [reply apps/files/templates/appnavigation.php]: okay, this is done 👍

**Review by kesselb — CHANGES_REQUESTED** [review]
Do not work like that.

`print_unescaped` expects a string. Right now a string, string and array are passed to this method. You are probably looking for something like:

 `print_unescaped('<a href="' . link_to_docs('user-webdav') . '" target="_blank" rel="noreferrer noopener">' . $l->t('Use this address to access your Files via WebDAV') . '</a>')`.

Luckily we can simplify that some more. That actual purpose of print_unescaped is to echo strings from a untrusted source (like transifex) unescaped. As we split markup and text now we don't need print_unescaped anymore:

`<em><a href="<?php echo link_to_docs('user-webdav') ?>" target="_blank" rel="noreferrer noopener"><?php p($l->t('Use this address to access your Files via WebDAV')) ?> ↗</a></em>`

**Comment by szaimen** [comment]
@kesselb thanks for the advice!
Would you mind to create a PR with that? 
I initially only wanted to put a bit focus on that special part of the code with my PR...

**Comment by kesselb** [comment]
Feel free to take the code and submit a pr.

**Comment by szaimen** [comment]
Thanks, I'll make a new one.

**Comment by szaimen** [comment]
superseded by #20186

**Pull request closed without merging**